### ISSUE TITLE
Text encoding of hashes

### ISSUE THREAD

**Issue by rklaehn** [n0-computer/iroh#1330]
I hate to bring this up again, but, can we revisit base32 encoding for hashes?

5 bits per char is not *that* much better than 4 bits per char, and I find it attractive if the hashes we show in the console gui are the same as what you get from `b3sum`...

So can we do hex, either as an alternative format ( {:#} ) or as a default? I think you should be able to write a parse fn that can distinguish between the two encodings.

**Comment by rklaehn** [comment]
Hex is also (correctly) the debug output of blake3::Hash and therefore our iroh-bytes::Hash. So you get a debug msg like this:

```
Hash(Hash("a75ec90437c976ec7277bac0f6db657d93e33a2a0f6a60f6ce8b8402423bdaea"))
```

and can't really use this as input for a CLI tool...

WDYT @b5?

**Comment by dignifiedquire** [comment]
> I find it attractive if the hashes we show in the console gui are the same as what you get from `b3sum`

with the CID prefixes that will never be the case though

**Comment by rklaehn** [comment]
> with the CID prefixes that will never be the case though

Yeah, but we are mostly dealing with raw hashes, so what are the cid prefixes good for again?

In any case I would love to be able to do `iroh get <a blake3 hash in hex format>` .

**Comment by dignifiedquire** [comment]
input should be easy, just try base32 and the. hex decoding

**Comment by b5** [comment]
I'm a fan of _additionally_ accepting a blake3 hash in hex format, but don't want to deviate from base32 right now.

> Yeah, but we are mostly dealing with raw hashes, so what are the cid prefixes good for again?

They're what make us an IPFS system. This would hurt a lot of long term plans for convergence between iroh & kubo. Sorry, while alignment with the output of `b3sum` is nice, it's not a compelling enough reason to stop being an IPFS system, which is what this suggest change would amount to

**Comment by b5** [comment]
Should we re-purpose this issue to support b3sum output as _input_?

**Comment by rklaehn** [comment]
https://github.com/n0-computer/iroh/pull/1590

**Comment by ramfox** [comment]
@rklaehn would you consider this closed with #1590 ?

**Comment by rklaehn** [comment]
Yeah, sorry.
